### DISCUSSION TITLE
Proprietary Application Licensing with Tauri

### DISCUSSION THREAD

**Discussion by MattTheCuber** [tauri-apps/tauri#11536] [Q&A]
My company is a developing proprietary software for profit using Tauri. Tauri's license supports this behavior under the MIT and Apache license. However, my questions lies with the Microsoft Visual Studio C++ Build Tools dependency.

The license for Microsoft Visual Studio C++ Build Tools is very complicated and confusing. It seems that they may be asking organizations to purchase a license to use the software. Since we are not directly using the build tools, does this requirement still fall on our company?

https://learn.microsoft.com/en-us/answers/questions/1683872/is-microsoft-c-build-tools-visual-studio-free-for

**Comment by FabianLars** [comment]
This indeed seems to be a tricky subject eg https://internals.rust-lang.org/t/visual-studio-build-tools-licensing/17097/5 / https://internals.rust-lang.org/t/use-lld-by-default-on-windows-to-mitigate-microsoft-c-build-tools-licencing-issues/16606/9 and it seems like it indeed requires getting the proper license if your project is not open source.

If all your rust crates support it, using the mingw toolchain instead of the msvc one may be worth to consider.

**Reply by Think-M** [reply]
How do I configure the Tauri project if I switch to compiling with the MinGW toolchain?

**Reply by strottos** [reply]
You can do this and this is what we've ended up doing, I think I just ran (or something like this):
```
rustup toolchain install stable-x86_64-pc-windows-gnu
rustup default stable-x86_4-pc-windows-gnu
```
and install mingw and it seems to be working fine, can even build nsis packages so long as I'm on the latest `tauri-cli`.

**Comment by MattTheCuber** [comment]
https://visualstudio.microsoft.com/license-terms/vs2022-ga-diagnosticbuildtools/

**Comment by strottos** [comment]
Just hit exactly the same issue myself. It's very unclear and it seems absurd to have to pay given clause 1d in the licence, clause 1d says:
```
Separately, if you don’t have a valid license to any of the Visual Studio Products you may still install and use copies of the software to compile and build C++ components that (i) have been released by a third party under an open-source software license approved by the Open Source Initiative; and (ii) are also reasonably required to build your applications (“Open Source Dependencies”).
```
You could argue this two ways I feel:
1. We're only using the linker, and it's completely non-sensical for the linker not to be included in the above as compiling C++ is futile without linking it into something in the end, so logically surely this should be included.
2. However it's not explicitly stated that you can use the linker for free.

I will really resent it if we have to pay for this just to use the linker when everything else is open source, but I feel like maybe that's what it says. Technically though that would mean nobody could use 1d above to build anything beyond an object file, which is basically useless on its own.

**Comment by strottos** [comment]
I should say we've one down the route of just using the mingw linker and it seems to be working fine. No licencing issues any more, haven't noticed any particular performance differences in the end product (though a touch slower to compile maybe but worth the hit vs paying for a licence just for a linker). First time I tried compiling `nsis` packages was failing but then when I upgraded to latest Tauri CLI that started working fine.

**Reply by Think-M** [reply]
I think the official website's documentation needs to be revised. It is still recommended to use the documentation for now:
`pnpm tauri build --target i686-pc-windows-msvc`
It should be indicated that there is a risk associated with this parameter and it is recommended to use GNU instead. I believe most people are not aware of this risk.

**Reply by strottos** [reply]
Totally agree
